charts/utils/templates/deployments.yaml

```diff
@@ -0,0 +1,59 @@
+{{ if .Values.deployments.entities }}
+{{ $prefix := printf "%s-%s" $.Values.client $.Values.environment }}
+{{- range $deployment := .Values.deployments.entities }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $prefix }}-{{ $deployment.name }}-client
+  {{- if $.Values.namespace }}
+  namespace: {{ $.Values.namespace }}
+  {{- end }}
+  labels:
+    app: {{ $prefix }}-{{ $deployment.name }}-client
```

Comment:
I think you can define blocks like `{{ $prefix }}-{{ $deployment.name }}` as a variable and use `this variable` everywhere to not repeat it few times (the chance of error would be less then)